typing/typedecl.ml

```diff
@@ -1075,7 +1113,9 @@ let rec check_constraints_rec env loc visited ty =
            *already* violate the constraints -- we need to report a problem with
            the unexpanded types, or we get errors that talk about the same type
            twice.  This is generally true for constraint errors. *)
-        match Ctype.matches ~expand_error_trace:false env ty ty' with
+        match Ctype.matches ~expand_error_trace:false env
+                (Ctype.instance ty) ty'
+        with
```

Comment:
Probably this should be posted upstream, too.

Reply:
https://github.com/ocaml/ocaml/pull/13725